### PR TITLE
Logging improvements

### PR DESCRIPTION
**Pull request by kissgyorgy**

### CODE REVIEW

File: unblob/logging.py [hunk 1]
```diff
@@ -2,22 +2,45 @@
...
+            elif isinstance(value, cstruct):
+                event_dict[key] = dumpstruct(value)
+
         return event_dict
 
-    return add_path
+    return convert_type
+
+
+class LazyField:
```
Comment (kukovecz) [inline unblob/logging.py:28]: I believe this (commit: 97f5700fea3f4e4805f0c9f9e77e2ac2e745f54a) was left here by mistake?

File: unblob/logging.py [hunk 2]
```diff
@@ -2,22 +2,45 @@
...
-    def add_path(logger, method_name: str, event_dict: structlog.types.EventDict):
+def pretty_print_types(extract_root: Path):
+    def convert_type(logger, method_name: str, event_dict: structlog.types.EventDict):
         use_absolute_path = event_dict.pop("_absolute_path", False)
         for key, value in event_dict.items():
             if isinstance(value, Path):
                 path = value if use_absolute_path else value.relative_to(extract_root)
                 event_dict[key] = str(path)
+
+            elif isinstance(value, cstruct):
```
Comment (kukovecz) [inline unblob/logging.py:20]: I really, really love this idea!

However, I wanted to see for myself, and it was not working. I check the `zip` handler's debug logs, they were in a "normal" format, so the `dumpstruct` was not running for them.

With this modification I could make it work:
```suggestion
            elif isinstance(value, cstruct) or isinstance(value, Instance):
```

Comment (kukovecz) [inline unblob/logging.py:20]: I really, really love this idea!

However, I wanted to see for myself, and it was not working. I checked the `zip` handler's debug logs, they were in a "normal" format, so the `dumpstruct` was not running for them.

With this modification I could make it work:
```python
from dissect.cstruct.types.instance import Instance
...

elif isinstance(value, cstruct) or isinstance(value, Instance):
```

**Review by kukovecz — CHANGES_REQUESTED** [review]
Here are my suggestions and I just clicked "Request changes" to see what will happen with the pull request this way :sweat_smile:

File: unblob/logging.py (hunk 1 above)
Comment (kissgyorgy) [reply unblob/logging.py:28]: I wanted to keep this in the history and reverted this commit.

Comment (kukovecz) [reply unblob/logging.py:28]: But you reverted the reverted commit, and it is in there, right now, and it should not be.

File: unblob/logging.py (hunk 2 above)
Comment (kissgyorgy) [reply unblob/logging.py]: Thanks, I changed it to check only for `Instance`, because it seems the parser methods will emit the instance of that.

**Pull request merged**